main/src/ca/uwaterloo/flix/api/lsp/Visitor.scala

```diff
@@ -97,6 +97,65 @@ object Visitor {
     def consumeTypeParam(tparam: TypeParam): Unit = ()
   }
 
+  case class StackConsumer() extends Consumer {
```

Comment:
I think this should go in a separate file.

Probably we should also more the Consumer and Acceptor to separate files.

Reply:
Also need some documentation of how this works.